tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -1074,6 +1112,42 @@ def test_search_registered_models(self):
             ([names[4]], None),
         )
 
+    def test_search_registered_models_by_tag(self):
+        name1 = "test_for_search_RM_by_tag1"
+        name2 = "test_for_search_RM_by_tag2"
+        tags1 = [
+            RegisteredModelTag("t1", "abc"),
+            RegisteredModelTag("t2", "xyz"),
+        ]
+        tags2 = [
+            RegisteredModelTag("t1", "abcd"),
+            RegisteredModelTag("t2", "xyz123"),
+            RegisteredModelTag("t3", "XYZ"),
+        ]
+        self._rm_maker(name1, tags1)
+        self._rm_maker(name2, tags2)
+
+        rms, _ = self._search_registered_models(f"tag.t3 = 'XYZ'")
+        self.assertEqual(set(rms), {name2})
```

Comment:
I don't think we should use `set` because the order matters `search_registered_models`.

Reply:
`test_search_registered_models` doesn't use `set`.

Reply:
What if `rms` contains duplicates? The test would pass, but it should not.

Reply:
test updated, now comparing result with list.